src/components/ThemeSwitcher.js

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import IconButton from '@mui/material/IconButton';
+import Box from '@mui/material/Box';
+import { useTheme, createTheme, ThemeProvider } from '@mui/material/styles';
+import Brightness4Icon from '@mui/icons-material/Brightness4';
+import Brightness7Icon from '@mui/icons-material/Brightness7';
+
+function ThemeSwitcher({ toggleDarkMode }) {
+  const theme = useTheme();
+
+  const modifiedDarkTheme = createTheme(theme, {
+    palette: {
+      mode: 'dark',
+      primary: {
+        main: '#2196f3',
+      },
+      secondary: {
+        main: '#ff4081',
+      },
+    },
+  });
+
+  return (
+    <ThemeProvider theme={modifiedDarkTheme}>
```

Comment:
This `ThemeProvider` component and the `modifiedDarkTheme` that you're passing it; will only affect the _children_ of this component. So in this case, the `primary`, and `secondary` palette values that you've chosen will only affect the `Box`, `IconButton`, and BrightnessIcon elements you have here; but nothing elsewhere in the app or page

Reply:
If we want to change the `primary.main` and `secondary.main` values of the overall theme; we need to add conditional values up in `App.js` where the `ThemeProvider` component is wrapping the _entire application_. 

Right now, when `toggleDarkMode` is called here in this component, it will cause the `mode` state variable in `App.js` to be changed. 

What we can do then is if you want to completely customize your themes based on the light/dark mode, is something like:

```jsx
// in App.js
  const [mode, setMode] = useState('light');

  const toggleDarkMode = (value) => {
    setMode(value);
  };

  const lightTheme = createTheme({
    palette: {
      mode: 'light',
      primary: { main: '#3f51b5' }
      // etc any other customizations that we want to make for light mode
    }
  });

  const darkTheme = createTheme({
    palette: {
      mode: 'dark',
      primary: { main: '#2196f3' }
      // etc any other customizations that you want to make for dark mode
    }
  });

  // set the currentTheme variable based on the `mode` state variable which may be toggled with the `toggleDarkMode` function
  const currentTheme =  mode === 'dark' ? darkTheme : lightTheme;

    return (
      <ThemeProvider theme={theme}>
      // etc
    );
}

Reply:
Thank you Akos , I have implemented your code 👍